src/app/components/Toolbar/Features/SearchAddress/SearchBox.tsx

```diff
@@ -0,0 +1,51 @@
+import { Box, Button, TextInput } from 'grommet'
+import { FormClose, Search } from 'grommet-icons/icons'
+import React from 'react'
+
+interface Props {
+  placeholder: string
+  value: string
+  onChange: (value: string) => void
+  onClear: () => void
+}
+
+/**
+ * Rounded input with search icon and clear button.
+ */
+export const SearchBox = React.forwardRef<HTMLInputElement, Props>((props, ref) => {
+  const { placeholder, value, onChange, onClear, ...rest } = props
+  const borderRadius = '30px'
+  return (
+    <Box
+      round={borderRadius}
+      border={{ size: '1px' }}
+      fill="vertical"
+      justify="center"
+      align="center"
+      width={{ max: '600px' }}
+      direction="row"
+    >
+      <TextInput
+        style={{ borderRadius: borderRadius }}
+        plain
+        icon={<Search />}
+        value={value}
+        onChange={e => onChange(e.target.value)}
+        placeholder={
+          <Box margin={{ left: '36px' }} max-height="30px" style={{ textOverflow: 'ellipsis' }} flex="shrink">
+            {placeholder}
+          </Box>
+        }
+        ref={ref}
+        {...rest}
+      />
+      {value && (
+        <Button
+          style={{ marginLeft: '-48px', zIndex: 1, marginTop: '-2px', marginBottom: '-2px' }}
```

Comment:
Two ideas:
- we can avoid hardcoding all these pixels here , so component will be more 'stable' when updating Grommet or theme in the future
- input height is not aligned with network selection button height, it's more visible on mobile view

something like this may work:

```
const FooButton = styled(Button)`
  margin-left: ${({ theme }) => `-${theme.global.edgeSize.large}`}; // avoid hardcoded pixels
  position: relative;
`
```

```
      <FooButton
        style={{
          visibility: value ? 'visible' : 'hidden',  // avoid jumping
        }}
        icon={<FormClose />}
        onClick={() => onClear()}
      />
```

Reply:
Oh yeah, that's better!

I didn't know grommet's theme is in styled-components. Though `theme.global.edgeSize.large` isn't related. 48px comes from "x" button's width: `left-padding + "x" icon width + right-padding`.

Is there any benefit of `position: relative;` vs `z-index: 1;`?

Reply:
`Is there any benefit of position: relative; vs z-index: 1;?` 
I have no idea 😄